src/main/scala/loamstream/loam/intake/aggregator/Metadata.scala

```diff
@@ -1,35 +1,200 @@
+        q <- quantitative
+      } yield {
+        Metadata(
+          dataset = dataset,
+          phenotype = ph,
+          varIdFormat = varIdFormat,
+          ancestry = ancestry,
+          author = author,
+          tech = tech,
+          quantitative = q)
+      }
+    }
+    
+    def toNoPhenotype: Try[NoPhenotype] = {
+      quantitative.map { q =>
+        NoPhenotype(
+          dataset = dataset,
+          varIdFormat = varIdFormat,
+          ancestry = ancestry,
+          author = author,
+          tech = tech,
+          quantitative = q)
+      }
+    }
+    
+    def toNoPhenotypeOrQuantitative: NoPhenotypeOrQuantitative = {
+      NoPhenotypeOrQuantitative(
+        dataset = dataset,
+        varIdFormat = varIdFormat,
+        ancestry = ancestry,
+        author = author,
+        tech = tech)
+    }
+  }
+  
+  override def fromConfig(config: Config): Try[Metadata] = {
+    import net.ceedubs.ficus.Ficus._
+    import net.ceedubs.ficus.readers.ArbitraryTypeReader._
+    
+    //NB: Marshal the contents of loamstream.intake.metadata into a Metadata instance.
+    //Names of fields in Metadata and keys under loamstream.intake.metadata must match.
+    Try(config.as[Parsed](Defaults.configKey)).flatMap(_.toMetadata)
+  }
+  
+  object Defaults {
+    val configKey: String = "loamstream.aggregator.intake.metadata"
+    val varIdFormat: String = "{chrom}_{pos}_{ref}_{alt}"
+  }
+  
+  final case class NoPhenotype(
+      dataset: String,
+      varIdFormat: String = Defaults.varIdFormat,
+      ancestry: String,
+      author: Option[String],
+      tech: String,
+      quantitative: Quantitative) {
+    
+    def toMetadata(phenotype: String): Metadata = {
+      Metadata(
+          dataset, 
+          phenotype, 
+          varIdFormat, 
+          ancestry, 
+          author, 
+          tech, 
+          quantitative = quantitative)
+    }
+  }
+  
+  object NoPhenotype extends ConfigParser[NoPhenotype] {
+    override def fromConfig(config: Config): Try[NoPhenotype] = {
+      import net.ceedubs.ficus.Ficus._
+      import net.ceedubs.ficus.readers.ArbitraryTypeReader._
+      
+      //NB: Marshal the contents of loamstream.intake.metadata into a Metadata instance.
+      //Names of fields in Metadata and keys under loamstream.intake.metadata must match.
+      Try(config.as[Parsed](Defaults.configKey)).flatMap(_.toNoPhenotype)
+    }
+  }
+  
+  final case class NoPhenotypeOrQuantitative(
+      dataset: String,
+      varIdFormat: String = Defaults.varIdFormat,
+      ancestry: String,
+      author: Option[String],
+      tech: String) {
+    
+    def toMetadata(phenotype: String, quantitative: Quantitative): Metadata = {
+      Metadata(
+          dataset, 
+          phenotype, 
+          varIdFormat, 
+          ancestry, 
+          author, 
+          tech, 
+          quantitative = quantitative)
+    }
+  }
+  
+  object NoPhenotypeOrQuantitative extends ConfigParser[NoPhenotypeOrQuantitative] {
```

Comment:
Same question as the `NoPhenotype`. The `subjects` is required. Is this for error reporting or just allowing someone to make the intake script w/o those settings for testing purposes?

Reply:
Not testing, but to allow something like:
```scala
val commonMetadata = ...
val config = Config.parseFile(configFile)
for {
  phenotype <- config
  quantitative <- getFromConfigFor(phenotype)
} {
  val actualMetadata = commonMetadata + phenotype + quantitative

  processDataAndUpload(actualMetadata, ...)
}
```